src/riakc_pb_socket.erl

```diff
@@ -3286,8 +3300,8 @@ live_node_tests() ->
                     {ok, Obj1} = ?MODULE:put(Pid, PO, [return_body]),
                     {ok, Obj2} = ?MODULE:put(Pid, PO, [return_body]),
                     %% Make sure the same key isn't generated twice
-                    ?assertEqual(element(1, Obj1), riakc_obj),
-                    ?assertEqual(element(1, Obj2), riakc_obj),
+                    ?assertEqual(riakc_obj, element(1, Obj1)),
+                    ?assertEqual(riakc_obj, element(1, Obj2)),
```

Comment:
Docs say the order should be switched:

http://erlang.org/doc/apps/eunit/chapter.html#Assert_macros